### PR TITLE
Settings: Performance Preferences

### PR DESCRIPTION
**Pull request by raghavaggarwal99**
Signed-off-by: raghavaggarwal99 <ragha170103048@iitg.ac.in>

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for submitting this PR, @raghavaggarwal99. Here's an issue that I bumped into while testing. This issue occurs when I navigate to /performance:

Screen Shot 2020-01-27 at 6 37 06 PM

The following log is produced from `meshery`:
```
Could not open MesheryPerformanceComponent.js in the editor.
The editor process exited with an error: spawn code ENOENT.

[ event ] disposing inactive page(s): /, /next/dist/pages/_error, /settings, /results
```

File: ui/components/MesherySettings.js [hunk 1]
```diff
@@ -137,6 +138,11 @@ class MesherySettings extends React.Component {
               <FontAwesomeIcon icon={faPoll} transform={mainIconScale} fixedWidth />
             } label="Metrics" />
           </Tooltip>
+          <Tooltip title="Performance" placement="top">
+            <Tab icon={
+              <FontAwesomeIcon icon={faPoll} transform={mainIconScale} fixedWidth />
```
Comment (leecalcote) [inline ui/components/MesherySettings.js]: Here's an icon that we've created for this tab - https://github.com/layer5io/meshery/blob/master/ui/static/img/performance.svg

Comment (raghavaggarwal99) [reply ui/components/MesherySettings.js]: this is done!

Comment (raghavaggarwal99) [reply ui/components/MesherySettings.js]: Though i don't know how to solve the above issue. I am facing the same one for a while now

Comment (leecalcote) [reply ui/components/MesherySettings.js]: @Nikhil-Ladha suggestions here?

File: handlers/load_test_preferences_handler.go [hunk 2]
```diff
@@ -25,17 +27,23 @@ func (h *Handler) LoadTestPrefencesHandler(w http.ResponseWriter, req *http.Requ
 		http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
 		return
 	}
+
 	if qps < 0 {
 		http.Error(w, "please provide a valid value for qps", http.StatusBadRequest)
 		return
 	}
+
+	fmt.Println(qps)
```
Comment (girishranganathan) [inline handlers/load_test_preferences_handler.go]: @raghavaggarwal99 I hope you will be removing the `fmt.Println` statements.

File: ui/components/MesheryPerformanceComponent.js [hunk 3]
```diff
@@ -193,6 +193,14 @@ class MesheryPerformanceComponent extends React.Component {
             </IconButton>
           ),
         });
+        self.props.updateLoadTestPref({loadTestPref: {
+          qps,
+          c,
+          t, 
+          gen,
```
Comment (girishranganathan) [inline ui/components/MesheryPerformanceComponent.js]: I am seeing an error `gen` is not defined. Can you please fix that reference?

File: ui/components/MesherySettingsPerformance.js [hunk 4]
```diff
@@ -0,0 +1,284 @@
+import React from 'react';
```
Comment (girishranganathan) [inline ui/components/MesherySettingsPerformance.js]: I presume this component is not complete.

**Comment by girishranganathan** [comment]
interestingly I don't see the issue Lee is facing...

**Comment by girishranganathan** [comment]
@raghavaggarwal99 @leecalcote fix for the issue @leecalcote is seeing is in this PR: https://github.com/layer5io/meshery/pull/658

File: ui/components/MesherySettings.js (hunk 1 above)
Comment (Nikhil-Ladha) [reply ui/components/MesherySettings.js]: Actually, I haven't yet tested it on my local. So, once i have done that I will share my feedback.

**Comment by kushthedude** [comment]
@raghavaggarwal99 Are you still willing to complete it ?
@leecalcote What is the motive of this PR ? As the code is still pre-mature and I don't see any relevant issue linked to the same ?

**Comment by leecalcote** [comment]
@kushthedude good question. @raghavaggarwal99 is on hiatus from the project for the moment. Here are the relevant issues:

feature - https://github.com/layer5io/meshery/issues/598
bug - https://github.com/layer5io/meshery/issues/676

**Comment by leecalcote** [comment]
@nveenjain @shivaylamba @Nikhil-Ladha do we know if this is a front-end or backend issue?

**Comment by Nikhil-Ladha** [comment]
Mostly it is backend but I guess only a small change in the front end is needed.

**Comment by shivaylamba** [comment]
On the front end we just need to improve the UI, but I guess it's more of an issue of not being properly saved in the back end since the network response is going, but it is not storing the User Preference

**Comment by kushthedude** [comment]
> On the front end we just need to improve the UI, but I guess it's more of an issue of not being properly saved in the back end since the network response is going, but it is not storing the User Preference

It is Frontend, Server is working properly. There is no response to being changes in the Data-Fetch. I also told @Nikhil-Ladha about this issue when User-Preference section was implemented by him.

**Comment by kushthedude** [comment]
I will open the fix for #676 . Meanwhile this PR is abandoned, it would be better to close this.

**Pull request closed without merging**